be/src/storage/rowset/beta_rowset_writer.cpp

```diff
@@ -329,7 +329,8 @@ Status HorizontalBetaRowsetWriter::flush_chunk_with_deletes(const vectorized::Ch
         RETURN_IF_ERROR(flush_del_file(deletes));
         return _flush_chunk(upserts);
     } else {
-        return flush_chunk(upserts);
+        /// There is impossible that both upserts and deletes are empty when flush memtable
+        __builtin_unreachable();
```

Comment:
Are you encounter some bug relating to this?  If not, I don't think it's favorable to reduce this function's compatibility, it's better to support the scenario that both upsert and deletes are empty rather than reduce this function's 'functionality'

Reply:
Another choice is just `return Status::OK();` when both them are empty.

Reply:
Are you encounter some bug relating to this? I mean if upserts is empty, the current code crashes or has unexpected behavior?

Reply:
No